.travis.yml

```diff
@@ -1,39 +1,33 @@
-sudo: required
+sudo: false
 dist: trusty
 language: node_js
 git:
-  depth: 10
+  depth: 3
 node_js:
   - "4"
   - "6"
 before_install:
-  # Remove ./node_modules/.bin from PATH so node-which doesn't replace Unix which and cause RVM to barf. See https://github.com/travis-ci/travis-ci/issues/5092
-  - export PATH=$(python -c 'from sys import argv;from collections import OrderedDict as od;print(":".join(od((p,None) for p in argv[1].split(":") if p.startswith("/")).keys()))' "$PATH")
-  - rvm install 2.2
-  - rvm use 2.2 --fuzzy
-  - npm install -g npm@3
-  - "export TRAVIS_COMMIT_MSG=\"$(git log --format=%B --no-merges -n 1)\""
+  - if [[ `npm -v` != 3* ]]; then npm install -g npm@3; fi
+  - "export TRAVIS_COMMIT_MSG=\"`git log --format=%B --no-merges -n 1`\""
   - echo "$TRAVIS_COMMIT_MSG" | grep '\[skip validator\]'; export TWBS_DO_VALIDATOR=$?; true
   - echo "$TRAVIS_COMMIT_MSG" | grep '\[skip sauce\]'; export TWBS_DO_SAUCE=$?; true
   - if [ "$TRAVIS_REPO_SLUG" = twbs-savage/bootstrap ]; then export TWBS_DO_VALIDATOR=0; fi
 install:
-  - bundle install --deployment --jobs=3
+  - bundle install --deployment --jobs=3 --retry=3
   - cp grunt/npm-shrinkwrap.json ./
-  - npm install
+  - npm install --only=development
+after_success:
+  - grunt/upload-preview.sh
 cache:
   directories:
     - node_modules
     - vendor/bundle
     - "$HOME/google-cloud-sdk"
 env:
-  global:
-    - NPM_CONFIG_PROGRESS="false"
-  matrix:
-    - TWBS_TEST=core
-    - TWBS_TEST=validate-html
-    - TWBS_TEST=sauce-js-unit
+  - TWBS_TEST=core
+  - TWBS_TEST=validate-html
+  - TWBS_TEST=sauce-js-unit
 matrix:
-  fast_finish: true
```

Comment:
[`fast_finish`](https://blog.travis-ci.com/2013-11-27-fast-finishing-builds/) only seems to have meaning where there are [allowed failures](https://docs.travis-ci.com/user/customizing-the-build/#Rows-that-are-Allowed-to-Fail).